llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -27420,10 +27420,8 @@ SDValue DAGCombiner::XformToShuffleWithZero(SDNode *N) {
       }
 
       APInt Bits;
-      if (auto *Cst = dyn_cast<ConstantSDNode>(Elt))
-        Bits = Cst->getAPIntValue();
-      else if (auto *CstFP = dyn_cast<ConstantFPSDNode>(Elt))
-        Bits = CstFP->getValueAPF().bitcastToAPInt();
+      if (auto OptBits = Elt->bitcastToAPInt())
+        Bits = *std::move(OptBits);
```

Comment:
Shouldn't this be std::move(*OptBits)? You want to move the APInt inside the optional.

Reply:
I don't think this makes much difference. `operator*` has [an r-value overload](https://en.cppreference.com/w/cpp/utility/optional/operator*).

But:
1. [cppreference](https://en.cppreference.com/w/cpp/utility/optional/operator*) has an `*std::move(opt2)` example.
2. I read `std::move` as "I will no longer use this variable", so it makes more sense to me on a variable than on the result of `operator*`.

Reply:
Ok. I didn't know there was an r-value overload for operator*. Thanks.